### PR TITLE
[MAINTENANCE] Statistics retention enhancements

### PR DESCRIPTION
**Pull request by lukasz-bigorajski**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/db/timeseries/questdb/QuestDbFEStatisticsRepository.scala [hunk 1]
```diff
@@ -82,6 +82,7 @@ private class QuestDbFEStatisticsRepository(private val engine: AtomicReference[
         name -> count
       }
       .toMap
+    logger.debug("Fetched FE statistics: {}", result)
```
Comment (coutoPL) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/db/timeseries/questdb/QuestDbFEStatisticsRepository.scala:85]: I'm not sure if we need it here. It seems that when we enable debug logging for a slick we will be able to see the same result. 

But if we see a need to have some better visibility of what happens in the statistics logic, I would consider adding some business-related debug logs at the `UsageStatisticsReportsSettingsService` level.

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/db/timeseries/questdb/RetentionTask.scala [hunk 2]
```diff
@@ -39,19 +36,28 @@ private[questdb] class RetentionTask(
...
+      .select(selectTruncatedToday, sqlContext)
+      .fetch(sqlContext) { record =>
+        record.getTimestamp(0)
+      }
+      .headOption
+
   // This check is needed because if the retention task would drop last partition, the db can't create new partition
   private def ensureOnePartitionExist(
-      clock: Clock,
+      maybeTruncatedToday: Option[Long],
```
Comment (coutoPL) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/db/timeseries/questdb/RetentionTask.scala:52]: I'm not sure if I understand what's inside the `maybeTruncatedToday`

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/db/timeseries/questdb/QuestDbFEStatisticsRepository.scala (hunk 1 above)
Comment (lukasz-bigorajski) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/db/timeseries/questdb/QuestDbFEStatisticsRepository.scala:85]: I forgot it to remove

**Review by coutoPL — APPROVED** [review]
LGTM

**Pull request merged**